lua/sp.c

```diff
@@ -2253,8 +2253,7 @@ static int lua_prepare_sql_int(SP sp, const char *sql, sqlite3_stmt **stmt,
 retry:
     memset(&err, 0, sizeof(struct errstat));
     if (sp->initial) {
-        sp->rc = get_prepared_stmt(sp->thd, sp->clnt, rec_ptr, &err, flags);
-        assert(sp->rc != SQLITE_PERM);
+        sp->rc = get_prepared_stmt_sp_init(sp->thd, sp->clnt, rec_ptr, &err, flags);
```

Comment:
Call `get_prepared_stmt_try_lock` with PREPARE_RECREATE

Reply:
And the other call site can invoke `get_prepared_stmt_try_lock` with  `flags ~ PREPARE_RECREATE`